lib/ldclient-rb/impl/datasystem.rb

```diff
@@ -0,0 +1,359 @@
+module LaunchDarkly
+  module Impl
+    module DataSystem
+      #
+      # This module contains the generic interfaces used for the data system (v1 and
+      # v2), as well as types for v1 and v2 specific protocols.
+      #
+      # @private
+      #
+
+      #
+      # Represents the availability of data in the SDK.
+      #
+      class DataAvailability
+        # The SDK has no data and will evaluate flags using the application-provided default values.
+        DEFAULTS = :defaults
+
+        # The SDK has data, not necessarily the latest, which will be used to evaluate flags.
+        CACHED = :cached
+
+        # The SDK has obtained, at least once, the latest known data from LaunchDarkly.
+        REFRESHED = :refreshed
+
+        ALL = [DEFAULTS, CACHED, REFRESHED].freeze
+
+        #
+        # Returns whether this availability level is **at least** as good as the other.
+        #
+        # @param [Symbol] self_level The current availability level
+        # @param [Symbol] other The other availability level to compare against
+        # @return [Boolean] true if this availability level is at least as good as the other
+        #
+        def self.at_least?(self_level, other)
+          return true if self_level == other
+          return true if self_level == REFRESHED
+          return true if self_level == CACHED && other == DEFAULTS
+
+          false
+        end
+      end
+
+      #
+      # Mixin that defines the required methods of a data system implementation. The data system
+      # is responsible for managing the SDK's data model, including storage, retrieval, and change
+      # detection for feature flag configurations.
+      #
+      # Application code should not need to implement this directly; it is used internally by the
+      # SDK's data system implementations.
+      #
+      module DataSystem
+        #
+        # Starts the data system.
+        #
+        # This method will return immediately. The system should signal when it has reached
```

Comment:
>  The system should signal

How? I think in python we required an even be provided. Are you doing something differently in ruby?